src/amber/cli/generators/generator.cr

```diff
@@ -3,6 +3,12 @@ require "./field.cr"
 
 module Amber::CLI
   abstract class Generator < Teeplate::FileTree
+    class_getter registered_commands = Hash(String, Amber::CLI::Generator.class).new
+
+    macro command(name)
```

Comment:
```cr
abstract class AuthBase
  #...

  macro inherited
    {% factory_id = @type.name.gsub(/^Amber::CLI::Generator::/, "").underscore %}

    ::Amber::CLI::Generator.registered_commands[{{factory_id.stringify}}] = self
  end
end
```

Reply:
@Sija yeah, i thought of doing that but currently some of the generators are abstract base classes and would cause an error if someone tried to run them

Reply:
There's still a limitation regarding naming the classes (`Amber::CLI::Generator::<CommandName>`), which doesn't exists for explicit declaration, so maybe it's better to keep the latter one avoiding potential confusion... just thinking out loud...